src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+
+
+struct RefinedTriangulation{Dc,Dp,A<:Triangulation{Dc,Dp},B<:RefinedDiscreteModel} <: Triangulation{Dc,Dp}
+  trian::A
+  model::B
+end
+
+function get_refined_model(t::RefinedTriangulation)
+  return t.model
+end
+
+# Wrap Triangulation API
+function Geometry.get_background_model(t::RefinedTriangulation)
+  get_background_model(t.trian)
+end
+
+function Geometry.get_grid(t::RefinedTriangulation)
+  get_grid(t.trian)
+end
+
+function Geometry.get_glue(t::RefinedTriangulation,::Val{d}) where d
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::RefinedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return RefinedTriangulation(v,t.model)
+end
+
+# Wrap constructors for RefinedDiscreteModel
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::RefinedTriangulation,args...;kwargs...)
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
```

Comment:
As far as I understand, I think that this condition might be too restrictive.  It is clear that if both triangulations are the same triangulation, `is_change_possible` must be true. However, I have the feeling that there would be more general cases in which this function must return true and it is currently returning false. For example 

```julia 
model              = CartesianDiscreteModel(...)
refined_model = refine(model)
strian = Triangulation(refined_model)
ttrian =  view(strian,1:2)
```

In this case, `is_change_possible` would return `false`, correct?

Reply:
I think that a possible way to to go here would be :

```julia 
if get_background_model(strian) === get_background_model(ttrian)
  return is_change_possible(strian.trian, ttrian)
else 
  ... # What you currently have below in your current version of the function
end
```

Reply:
Yes, I've been thinking a bout this. There is actually a todo just on top of the function that explains this. I also think that this might be the way to go, however I think we are missing one additional case: What if the triangulations are defined on parent-child models (we have refinement) AND we also have a change in triangulation type (like skeleton-normal or views, etc...). 
For instance, imagine we have a normal cell-triangulation on the child (fine) mesh and a skeleton triangulation on the parent (coarse) mesh (for instance, while doing something like domain decomposition). Then to transfer something from one to another we need to first change mesh and then change triangulation. How do we check for this? How do we do the change?

Reply:
Or should we just not care about this case? I.e we leave to the user the chore of doing two steps?

Reply:
For instance, consider the following example: 
```
model0 = ....
model1 = refine(model0)
model2 = refine(model1)

trian0 = some incompatible triangulation with a skeleton triangulation
trian1 = SkeletonTriangulation(model1)
trian2 = Triangulation(model2)
```

In this case, `is_change_possible(trian1,trian2)` would look at the parent/son relation between the models and return true. But it would do so without understanding that we have to do two steps to transform any `CellField` from onetriangulation to the other. 
Also, `is_change_possible(trian0,trian1)` would also return true because the father/son relation holds up. And this is a mistake! because even if we can move a function from one grid to another, the next step is impossible. 

So the question would be: Do we need to concern ourselves about this? Or should we return false in these cases and have this complexity be left to the user? 

@amartinhuertas

Reply:
After some talk, we considered that indeed `parent` should hold (if applicable) a `RefinedDiscreteModel` in order to be able to trace back the hierarchy.